graphrag/app/supportai/supportai.py

```diff
@@ -116,12 +119,122 @@ def init_supportai(conn: TigerGraphConnection, graphname: str) -> tuple[dict, di
+        # we generate random project name each time below
+        # Delete existing project if it exists
+        # existing_projects = bda_client.list_data_automation_projects()["projects"]
+        # for project in existing_projects:
+        #     if project["projectName"] == "barclays-preprocessing-project":
+        #         bda_client.delete_data_automation_project(projectArn=project["projectArn"])
+        #         time.sleep(2)
+        #         break
+
+        # Create BDA project
+        logger.info(f"Creating BDA project")
+        project_name = f"bda-preprocessing-{uuid.uuid4().hex[:6]}"
+        logger.info(f"Created before  BDA project {project_name}")   
+        project_response = bda_client.create_data_automation_project(
+            projectName=project_name,
+            projectDescription='Preprocessing multi data formats using bedrock data automation',
+            projectStage='DEVELOPMENT',
+            standardOutputConfiguration={
+                "document": {
+                    "extraction": {
+                        "granularity": {"types": ["DOCUMENT", "PAGE", "ELEMENT", "LINE", "WORD"]},
+                        "boundingBox": {"state": "ENABLED"}
+                    },
+                    "generativeField": {"state": "ENABLED"},
+                    "outputFormat": {
+                        "textFormat": {"types": ["PLAIN_TEXT", "MARKDOWN", "HTML", "CSV"]},
+                        "additionalFileFormat": {"state": "ENABLED"}
+                    }
+                }}
+        )
+        
+        project_arn = project_response['projectArn']
+        logger.info(f"Created after BDA project {project_name} with ARN {project_arn}")
+        # Get file names from S3
+        response = s3.list_objects_v2(Bucket=bucket_name)
+        objects = response.get('Contents', [])
+        logger.info(f"Found {len(objects)} objects in S3 bucket {bucket_name}")
+        job_results = []
+
+        # Create BDA job for each file
+        for obj in objects:
+            time.sleep(2)
+            file_key = obj['Key']
+            input_s3_uri = f's3://{bucket_name}/{file_key}'
+            output_s3_uri = f's3://{output_bucket}/bda/output/{file_key}'
+
+            bda_response = bda_runtime_client.invoke_data_automation_async(
+                inputConfiguration={
+                    's3Uri': input_s3_uri
+                },
+                outputConfiguration={
+                    's3Uri': output_s3_uri
+                },
+                dataAutomationConfiguration={
+                    'dataAutomationProjectArn': project_arn,
+                    'stage': 'DEVELOPMENT'
+                },
+                dataAutomationProfileArn='arn:aws:bedrock:us-west-2:734036167395:data-automation-profile/us.data-automation-v1'
+            )
+            logger.info(f"Created BDA job for file {file_key} with job ID {bda_response.get('invocationArn')}") 
+                
+                # Poll for job status
```

Comment:
This job status check loop should be moved out of the parent for loop to improve performance as BDA jobs are async.